pkg/drivers/bess/bess.go

```diff
@@ -348,10 +348,10 @@ func (d *Driver) createSwitch(module *network.Switch) error {
 		log.Errorf("Failed to set default gate for l2forward, error: %s", res.GetError().GetErrmsg())
 		return errors.New(res.GetError().GetErrmsg())
 	}
-    // We need to recover the original mod name from Switch_%s
-    orig_name := module.GetName() //expected as Switch_%s - from pipeline module
-    r := NewReader(orig_name[len("Switch_"):], port.SocketPath, d.notifications)
```

Comment:
There is still an issue here where this port name isn't right. The format of the port name will be "Switch_<meta network>_<port name>". An easier way to remove the meta portion is to just call module.GetMeta(). Here is the diff for a patch I used and it seems to be working (I see macs learned and no errors). Can you please update this part and then I think we are good to merge!
```
[trozet@trozet nimbess-agent]$ git diff
diff --git a/pkg/agent/agent.go b/pkg/agent/agent.go
index af17316..a394f57 100644
--- a/pkg/agent/agent.go
+++ b/pkg/agent/agent.go
@@ -699,7 +699,7 @@ func (s *NimbessAgent) processNotification() (bool) {
        pipeline, ok := s.Pipelines[fibRequest.Port]
 
        if (!ok) {
-               log.Errorf("Failed to find metaPipeline and its fib")
+               log.Errorf("Failed to find metaPipeline and its fib for port: %s", fibRequest.Port)
                return true /* This is bad, but it is not a reason to exit the notification thread */
        }
 
diff --git a/pkg/drivers/bess/bess.go b/pkg/drivers/bess/bess.go
index 2497933..beda457 100644
--- a/pkg/drivers/bess/bess.go
+++ b/pkg/drivers/bess/bess.go
@@ -349,8 +349,14 @@ func (d *Driver) createSwitch(module *network.Switch) error {
                return errors.New(res.GetError().GetErrmsg())
        }
        // We need to recover the original mod name from Switch_%s
-       orig_name := module.GetName() //expected as Switch_%s - from pipeline module
-       r := NewReader(orig_name[len("Switch_"):], port.SocketPath, d.notifications)
+       origName := module.GetName()
+       metaName := module.GetMeta() //expected as Switch_<meta name> - from pipeline module
+       if len(metaName) == 0 {
+               log.Warningf("Meta pipeline empty for Switch: %s", origName)
+       }
+       derivedPortName := origName[len(metaName)+1:] // remove "<MetaName>_" prefix
+       log.Debugf("Module name: %s, meta name: %s, port name: %s", origName, metaName, derivedPortName)
+       r := NewReader(derivedPortName, port.SocketPath, d.notifications)
        go r.Run()
        log.Infof("BESS Switch created with L2FWD: %s, Replicate: %s", l2Fwd.Name, rep.Name)
        return nil
```

Reply:
Sorry just realized github parsed my 2nd sentence and it is wrong in the comment. The format of the port name will be '''Switch_<meta network>'''

Reply:
Still parsing it haha so will just state it like this Switch_metaNetwork